### PR TITLE
🎉 octavia-cli: add pypi package workflow

### LINKED ISSUE #11394: 🐙 octavia-cli: deploy on Pypi (the python package index)
Python users want to `pip install octavia-cli`.

This will require:
* Checking which version of Python the project is compatible with.
* Making sure `setup.py` follows Pypi expectation
* Deploying the generated API client to Pypi too, as this is a dependency for the CLI.

### PR DESCRIPTION
**Pull request by dandpz**
## What
Octavia-cli is not available as python package

Resolves: #11394
Co-authored by @FabrizioCafolla

## How
Add workflow to publish python package to PyPI

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Comment by sh4sh** [comment]
Awesome, running checks in https://github.com/airbytehq/airbyte/pull/22671

@alafanechere octavia-cli pypi package implementation!

File: .github/workflows/gradle.yml
Comment (alafanechere) [inline .github/workflows/gradle.yml:168]: The publication to PyPI should happen when a new version is released, not on build.
I think it belongs to this [workflow](https://github.com/airbytehq/airbyte/blob/f94ef99f3d436434903f8c1357898b39423eaeab/.github/workflows/release-airbyte-os.yml#L72) instead. 
Moreover, these GitHub secrets are not defined on our repo so it won't work out of the box. 
Let's reuse the same action structure we use to publish the CDK ([here](https://github.com/airbytehq/airbyte/blob/master/.github/workflows/publish-cdk-command-manually.yml#L158) cc @maxi297 ) the action is different (mariamrf/py-package-publish-action@v1.1.0).

**Review by alafanechere — DISMISSED** [review]
Thanks for the initiative but I think a bit of rework is required 😄

**Comment by dandpz** [comment]
> Thanks for the initiative but I think a bit of rework is required 😄 

Sure, thanks for the review, I will follow your suggestions and update the pr 👍

File: .github/workflows/gradle.yml
Comment (sh4sh) [reply .github/workflows/gradle.yml]: These suggestions have been implemented since this comment. @alafanechere any further review?

**Review by sh4sh — APPROVED** [review]
I'd like to wait for @alafanechere's approval before merging, but this looks good to me.

**Comment by alafanechere** [comment]
I apologize for the delay but I'd like to hand this over to @rileybrook who's product managing the CLI now. I think distributing it to Pypi is a product decision.

**Comment by rileybrook** [comment]
@sh4sh  @dandpz Feel free to proceed 👍

**Pull request merged**
